src/main/java/org/apache/accumulo/examples/cli/ClientOpts.java

```diff
@@ -0,0 +1,143 @@
+    @Override
+    public ColumnVisibility convert(String value) {
+      return new ColumnVisibility(value);
+    }
+  }
+
+  public static class TimeConverter implements IStringConverter<Long> {
+    @Override
+    public Long convert(String value) {
+      if(value.matches("[0-9]+"))
+        value = "PT"+value+"S"; //if only numbers then assume seconds
+      return Duration.parse(value).toMillis();
+    }
+  }
+
+  public static class MemoryConverter implements IStringConverter<Long> {
+    @Override
+    public Long convert(String str) {
+      try {
+        char lastChar = str.charAt(str.length() - 1);
+        int multiplier = 0;
+        switch (Character.toUpperCase(lastChar)) {
+          case 'G':
+            multiplier += 10;
+          case 'M':
+            multiplier += 10;
+          case 'K':
+            multiplier += 10;
+          case 'B':
+            break;
+          default:
+            return Long.parseLong(str);
+        }
+        return Long.parseLong(str.substring(0, str.length() - 1)) << multiplier;
+      } catch (Exception ex) {
+        throw new IllegalArgumentException("The value '" + str + "' is not a valid memory setting. A valid value would a number "
+            + "possibily followed by an optional 'G', 'M', 'K', or 'B'.");
+      }
+    }
+  }
+
+  public static class PropertiesConverter implements IStringConverter<Configuration> {
+    @Override
+    public Configuration convert(String filename) {
+      try {
+        return new PropertiesConfiguration(filename);
+      } catch (ConfigurationException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Parameter(names = {"-c", "--conf"}, required = true, converter = PropertiesConverter.class,
+      description = "Config file for connecting to Accumulo.  See README.md for details.")
+  private Configuration config = null;
+
+  @Parameter(names = {"-auths", "--auths"}, converter = AuthConverter.class, description = "the authorizations to use when reading or writing")
+  public Authorizations auths = Authorizations.EMPTY;
+
+  public Connector getConnector() {
+    try {
+      ZooKeeperInstance zki = new ZooKeeperInstance(config);
+      return zki.getConnector(getPrincipal(), getToken());
+    } catch (AccumuloException | AccumuloSecurityException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public ClientConfiguration getClientConfiguration() {
+    return new ClientConfiguration(config);
+  }
+
+  public String getPrincipal() {
+    return config.getString("accumulo.examples.principal", "root");
+  }
+
+  public AuthenticationToken getToken() {
+    if (config.containsKey("instance.rpc.sasl.enabled")) {
+      try {
+        if (config.containsKey("accumulo.examples.keytab")) {
+          String keytab = config.getString("accumulo.examples.keytab");
+          return new KerberosToken(getPrincipal(), new File(keytab));
+        } else {
+          return new KerberosToken(getPrincipal());
```

Comment:
In IRC @joshelser also suggested the following :

> we want to invoke the example as the kerberos user, and the way to do that is we invoke some code in a doAs()
> if each example was a Runnable
> could change ClientOpts to have a method login(.., Runnable) method
> which could encapsulate that indirection

I think I am going to back out my incorrect Kerberos additions for now and keep the focus of this PR on using Accumulo's public API.   I'll look into a follow up PR for adding Kerberos to the examples.  @joshelser thanks for the feedback!